Reactions/ReactorBase.H

```diff
@@ -88,11 +89,18 @@ public:
     const amrex::Real absTol,
     const std::string& solvername);
 
+#if defined(AMREX_USE_HIP) || defined(AMREX_USE_CUDA)
+  ~ReactorBase() { amrex::The_Arena()->free(m_typ_vals_gpu); }
+#else
   ~ReactorBase() override = default;
+#endif
 
 protected:
   int verbose{0};
   amrex::Array<amrex::Real, NUM_SPECIES + 1> m_typ_vals = {0.0};
+#if defined(AMREX_USE_HIP) || defined(AMREX_USE_CUDA)
```

Comment:
So why don't we make `m_typ_vals` a `GpuArray` (since it is a fixed size array) and then we just carry that around instead of also carrying a `_gpu` version around as well? Then we don't need the alloc/free/copy from host? I must be missing something...

Reply:
I think you might be right. I haven't really looked into how the typical values are recomputed if so.

Reply:
Ok I think I did that... hopefully this works.